src/attr/_make.py

```diff
@@ -640,8 +640,13 @@ def slots_setstate(self, state):
             if not closure_cells:  # Catch None or the empty list.
                 continue
             for cell in closure_cells:
-                if cell.cell_contents is self._cls:
-                    set_closure_cell(cell, cls)
+                try:
+                    match = cell.cell_contents is self._cls
+                except ValueError:  # ValueError: Cell is empty
+                    pass
+                else:
+                    if match:
+                        set_closure_cell(cell, cls)
```

Comment:
Can't you just move this into the try block?

Reply:
I'd rather not - if a `ValueError` is raised by `set_closure_cell(...)` one day, this is maybe something you'd like to know asap, not after investigating a potential side-effect that may pop up somewhere else and take longer to be reported as an issue. But of course that's your decision, let me know.